uploadNeuroDB/file_insertion.pl

```diff
@@ -0,0 +1,509 @@
+
+# Generic TODOs:
+
+##TODO 1: once ExitCodes.pm class merged, replace exit codes by the variables
+# from that class
+
+
+use strict;
+use warnings;
+use Getopt::Tabular;
+use File::Basename;
+use File::Temp qw/ tempdir /;
+use Date::Parse;
+
+###### Import NeuroDB libraries to be used
+use NeuroDB::DBI;
+use NeuroDB::MRI;
+use NeuroDB::File;
+use NeuroDB::MRIProcessingUtility;
+##TODO 1: add line use NeuroDB::ExitCodes;
+
+
+##TODO 1: move those exit codes to ExitCodes.pm
+my $INVALID_SCANNER_ID     = 140; # new
+my $PROTOCOL_ID_NOT_FOUND  = 141; # new
+my $GET_SESSION_ID_FAILURE = 142; # new
+my $INVALID_UPLOAD_ID      = 143; # new
+my $GET_SUBJECT_ID_FAILURE = 96;  # same as MRIProcessingUtility.pm
+my $CANDIDATE_MISMATCH     = 111; # same as minc_insertion.pl
+my $FILE_NOT_UNIQUE        = 6;
+
+###### Table-driven argument parsing
+
+# Initialize variables for Getopt::Tabular
+my $profile       = undef;
+my $file_path     = undef;
+my $upload_id     = undef;
+my $patient_name  = undef;
+my $output_type   = undef;
+my $scan_type     = undef;
+my $date_acquired = undef;
+my $scanner_id    = undef;
+my $coordin_space = undef;
+my $metadata_file = undef;
+my $verbose       = 0;
+my $reckless      = 0;   # only for playing & testing. Don't set it to 1!!!
+my @args;
+
+# Describe the usage to be displayed by Getopt::Tabular
+my  $Usage  =   <<USAGE;
+
+This script inserts a file in the files and parameter_file tables.
+
+Usage: perl file_insertion.pl [options]
+
+-help for options
+
+USAGE
+
+# Set the variable descriptions to be used by Getopt::Tabular
+my $profile_desc       = "name of config file in ./dicom-archive/.loris_mri.";
+my $file_path_desc     = "file to register into the database (full path from "
+                         . "the root directory is required)";
+my $upload_id_desc     = "ID of the uploaded imaging archive containing the "
+                         . "file given as argument with -file_path option";
+my $pname_desc         = "patient name, if cannot be found in the file name "
+                         . "(in the form of PSCID_CandID_VisitLabel)";
+my $output_type_desc   = "file's output type (e.g. native, qc, processed...)";
+my $scan_type_desc     = "file's scan type (from the mri_scan_type table)";
+my $date_acquired_desc = "acquisition date for the file (YYYY-MM-DD)";
+my $scanner_id_desc    = "ID of the scanner stored in the mri_scanner table";
+my $coordin_space_desc = "Coordinate space of the file to register (e.g. "
+                         . "native, linear, nonlinear, nativeT1)";
+my $reckless_desc      = "upload data to database even if study protocol is "
+                         . "not defined or violated.";
+my $metadata_file_desc = "file that can be read to look for metadata "
+                         . "information to attach to the file to be inserted";
+
+
+# Initialize the arguments table
+my @args_table = (
```

Comment:
Super clear and neat way to declare the program arguments! I am making a mental note to myself for when I have to do the same thing.  :+1: